src/main/java/com/redhat/darcy/salesforce/CheckboxInputField.java

```diff
@@ -0,0 +1,112 @@
+/*
+ Copyright 2014 Red Hat, Inc. and/or its affiliates.
+
+ This file is part of darcy-salesforce.
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.redhat.darcy.salesforce;
+
+import static com.redhat.darcy.salesforce.RequiredInput.requiredInput;
+import static com.redhat.darcy.ui.Elements.checkbox;
+import static com.redhat.darcy.web.By.htmlTag;
+
+import com.redhat.darcy.ui.AbstractViewElement;
+import com.redhat.darcy.ui.annotations.Require;
+import com.redhat.darcy.ui.api.Locator;
+import com.redhat.darcy.ui.api.elements.Checkbox;
+import com.redhat.darcy.ui.api.elements.Element;
+import com.redhat.darcy.ui.api.elements.Requireable;
+
+/**
+ * An HTML input element for a value that corresponds to a checkbox field on 
+ * a Salesforce object.
+ */
+public class CheckboxInputField extends AbstractViewElement implements Checkbox,
+    Requireable {
+
+    @Require
+    private Element parent = super.parent;
```

Comment:
So--and the only reason I'm making a fuss really is because, being the first official library of elements for darcy, I'm cautious about setting the wrong precedent--I think I wouldn't use this pattern unless we explicitly have a reason to `@Require` the parent element. I also am probably going to go ahead and require it by default anyway, and then instead allow you to `@NotRequire` it if you absolutely have to for some reason, but I'll add that on an as-needed basis if it comes to it. (Actually I'm starting to lean towards requiring all element fields by default).